Automattic/wp-calypso
# Monetize: Update supporters list

## Proposed Changes

* Updates the Monetizes > Supporters list table. It now has several new columns which collectively show product title, product price and period, total paid, and date of first purchase. 
* This PR starts moving the Supporters screen toward the new Figma designs here: DqXCQr1dEWpF3P2dIEwiwd-fi-791_88014

**Before**
<img width="1091" alt="supporters-before" src="https://github.com/Automattic/wp-calypso/assets/21228350/91209cad-72b9-4639-8e3e-d8cda2767ca1">

**After**
<img width="1110" alt="supporters-after-2" src="https://github.com/Automattic/wp-calypso/assets/21228350/ea16d5c0-7226-412b-9b99-cf5d7f9e324d">

## Testing Instructions



1. You will need a WordPress simple site with stripe connected, some paid plans, and some actual paid purchases. The table in this list only shows paid customers. 
2. Go to http://calypso.localhost:3000/earn/supporters/YOURSITESLUG, and confirm the new table looks like the new screenshot above. 

## Pre-merge Checklist
